src/cart-button/index.ts

```diff
@@ -0,0 +1,80 @@
+import { tag, Events, ProductTransformer, Selectors, Tag, TransformUtils } from '@storefront/core';
+
+const DEFAULT_VALUE = 1;
+
+@tag('gb-cart-button', require('./index.html'))
+class CartButton {
+  state: CartButton.State = {
+    quantity: DEFAULT_VALUE
+  };
+
+  props: CartButton.Props;
+
+  // init() {
+    // keep this for service in core
+    // this.services.cart.register(this);
+  // }
+
+  onClick(event: MouseEvent & Tag.Event) {
+    event.preventUpdate = true;
+    if (this.props.onClick) {
+      this.props.onClick(event);
+    } else {
+      this.addItem(this.props.product, this.state.quantity);
+    }
+  }
+
+  addItem = (item: CartButton.RawProduct, quantity: number) => {
+    const transformed = this.productTransformer(item, quantity);
+
+    this.flux.store.dispatch(this.flux.actions.addToCart(transformed));
+  }
+
+  quantityHandler = (event: MouseEvent | TouchEvent) => {
+    this.set({ quantity: Number(event.target['value']) });
+  }
+
+  productTransformer = (item: CartButton.RawProduct, quantity: number): CartButton.CartProduct => {
+    const { structure } = this.config.cart;
+    const data: any = TransformUtils.remap(item, <any>structure);
+    data['collection'] = this.config.collection;
+    data['metadata'] = [{ key: 'image', value: data['image'] }];
+    data['quantity'] = quantity;
+    delete data['image'];
```

Comment:
If you just want to remove the image property from data, you can use this neat destructuring trick:
on line 39 write:
`const { image, ...allProps } = TransformUtils.remap(item, <any>structure);`
then you can make an object data:
```
const data = { 
  ...allProps,
  collection: this.config.collection,
  metadata: [{ key: 'image', value: data['image'] }],
  quantity,
};
```
I just found out about this trick, it's p neat!